src/tools/build_helper/src/git.rs

```diff
@@ -159,3 +159,37 @@ pub fn get_git_untracked_files(
         .collect();
     Ok(Some(files))
 }
+
+/// Print a warning if the branch returned from `updated_master_branch` is old
+///
+/// For certain configurations of git repository, this remote will not be
+/// updated when running `git pull`.
+///
+/// This can result in formatting thousands of files instead of a dozen,
+/// so we should warn the user something is wrong.
+pub fn warn_old_master_branch(
+    config: &GitConfig<'_>,
+    git_dir: &Path,
+) -> Result<(), Box<dyn std::error::Error>> {
+    use std::time::Duration;
+    const WARN_AFTER: Duration = Duration::from_secs(60 * 60 * 24 * 10);
+    let updated_master = updated_master_branch(config, Some(git_dir))?;
+    let branch_path = git_dir.join(".git/refs/remotes").join(&updated_master);
```

Comment:
This logic is incorrect for git worktrees, where `.git` is not a folder -- it is a file containing the path to where the .git folder lies.

Please either fix this PR or revert it, currently it causes incorrect warnings for everyone using worktrees.

Reply:
Ah, that has already been reported at https://github.com/rust-lang/rust/issues/130111. :)